File: editor/en/[1768]Merge Strings Alternately.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int len1 = word1.length();
        int len2 = word2.length();
        int i = 0;
        int j = 0;
        StringBuilder sb = new StringBuilder();
        while (i < len1 && j < len2) {
            sb.append(word1.charAt(i++));
            sb.append(word2.charAt(j++));
        }
        if (i < len1) {
            sb.append(word1.substring(i));
        }
        if (j < len2) {
            sb.append(word2.substring(j));
        }
        return sb.toString();
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1768]Merge Strings Alternately.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = []
        length1 = len(word1)
        length2 = len(word2)
        min_length = min(length1, length2)

        for i in range(min_length):
            result.append(word1[i])
            result.append(word2[i])

        result.extend(word1[min_length:])
        result.extend(word2[min_length:])
        return ''.join(result)

# leetcode submit region end(Prohibit modification and deletion)
